pipeline/dbt/models/staging/sources/mediation_numerique/stg_mediation_numerique__structures.sql

```diff
@@ -2,35 +2,58 @@ WITH source AS (
     {{ stg_source_header('mediation_numerique', 'structures') }}
 ),
 
+labels_nationaux_seed AS (
+    SELECT * FROM {{ ref('labels_nationaux') }}
+),
+
+labels_nationaux_prep AS (
+    SELECT
+        data ->> 'id'                                                         AS id,
+        JSONB_ARRAY_ELEMENTS_TEXT(NULLIF(data -> 'labels_nationaux', 'null')) AS labels_nationaux
+    FROM source
+),
+
+labels_nationaux AS (
+    SELECT
+        id,
+        ARRAY_AGG(DISTINCT labels_nationaux) AS labels_nationaux
+    FROM labels_nationaux_prep
+    WHERE
+        labels_nationaux IN (
+            SELECT labels_nationaux_seed."value" FROM labels_nationaux_seed
+        )
+    GROUP BY id
```

Comment:
yes, je me demandais si il y avait moyen de faire plus court avec un INNER JOIN au lieu du WHERE ... IN mais je n'ai pas forcément le temps de creuser pour voir si ça rendrait la chose plus claire :/ En attendant c'est assez explicite.

Reply:
Valentin m'a propose un INTERSECT et effectivement c'est exactement ce qu'on veux!